test/fw/ptl/utils/pbs_logutils.py

```diff
@@ -291,6 +291,13 @@ def open_log(self, log, hostname=None, sudo=False, start=None,
         taillogs = 10000
         tailcmd = [self.du.which(hostname, 'tail')]
         headcmd = [self.du.which(hostname, 'head')]
+        if sudo:
+            tailcmd = self.du.sudo_cmd + tailcmd
+            headcmd = self.du.sudo_cmd + headcmd
+        if hostname:
+            tailcmd = ['ssh', hostname] + tailcmd
+            headcmd = ['ssh', hostname] + headcmd
+
```

Comment:
hostname can be None.  Why don't you want to use self.du.head() and self.du.tail()?  Those take care of all of this for you.  You are basically reinventing the wheel here.